clang/lib/Driver/ToolChains/Arch/RISCV.cpp

```diff
@@ -173,14 +173,6 @@ void riscv::getRISCVTargetFeatures(const Driver &D, const llvm::Triple &Triple,
     }
   }
 
-  if (llvm::is_contained(Features, "+xcheri")) {
```

Comment:
Hm, I don't remember exactly why I added this before, though I have a funny feeling something about [+-]xcheri being a prefix of [+-x]cheri-rvc might have had something to do with it. Looks like `-march=rv64gc -mxcheri-rvc` might be broken as a result since RISCVTargetMachine is a bit sloppy in searching FS.

Reply:
Yeah that does look like it's a pre-existing bug. Will fix as part of updating the PR.